src/utils/Utils.ts

```diff
@@ -196,39 +196,56 @@ export class Utils {
     return value;
   }
 
-  static getFieldValue(result: IQueryResult, name: string): any {
-    // Be as forgiving as possible about the field name, since we expect
-    // user provided values.
-    if (name == null) {
-      return undefined;
-    }
-    name = Utils.trim(name);
-    if (name[0] == '@') {
-      name = name.substr(1);
-    }
-    if (name == '') {
+  /**
+   * Get the value of a field.
+   * If multiple names are provided the first result not null is returned.
+   *
+   * @param result a QueryResult in which to ge the fieldvalue.
+   * @param name One or multiple fieldNames to get the value.
+   */
+  static getFieldValue(result: IQueryResult, name: string | Array<string>): any {
+    if (Array.isArray(name)) {
+      for (let i = 0; i < name.length; i++) {
+        let value = this.getFieldValue(result, name[i]);
```

Comment:
Really not a fan of this. 

The name of the utility method now simply does not match what happens *inside* the method.

It's an "implicit" behaviour, where you have to read the method documentation to actually understand what happens when you use it, or read the implementation. The function now has 2 very different behaviour depending on the type of parameters you pass into it.

I would revert all of this.